ksqldb-engine/src/main/java/io/confluent/ksql/util/SandboxedSharedKafkaStreamsRuntimeImpl.java

```diff
@@ -71,9 +72,36 @@ public void register(
       final QueryId queryId
   ) {
     collocatedQueries.put(queryId, binpackedPersistentQueryMetadata);
-    if (!kafkaStreams.getTopologyByName(
-        binpackedPersistentQueryMetadata.getQueryId().toString()).isPresent()) {
-      kafkaStreams.addNamedTopology(binpackedPersistentQueryMetadata.getTopologyCopy(this));
+    try {
+      if (!kafkaStreams.getTopologyByName(
+          binpackedPersistentQueryMetadata.getQueryId().toString()).isPresent()) {
+        log.debug("Registering query {} for validation", queryId);
+        try {
+          kafkaStreams.addNamedTopology(binpackedPersistentQueryMetadata.getTopologyCopy(this))
+              .all()
+              .get();
+        } catch (final Throwable e) {
+          log.error(String.format("Validation for query %s failed due to:", queryId), e);
+          throw e;
+        }
+      } else {
+        log.info("Validation for a create or replace runtime");
+        try {
+          kafkaStreams.removeNamedTopology(binpackedPersistentQueryMetadata.getQueryId().toString())
+              .all()
+              .get();
+          kafkaStreams.addNamedTopology(binpackedPersistentQueryMetadata.getTopologyCopy(this))
+              .all()
+              .get();
+        } catch (final Throwable e) {
+          log.error(String.format("Validation for query %s failed due to:", queryId), e);
+          throw e;
+        }
+      }
+    } catch (ExecutionException | InterruptedException e) {
+      log.error(String.format(
+          "Failed to validate query %s within the allotted timeout due to",
```

Comment:
```suggestion
          "Failed to validate query %s due to",
```
There's no timeout for query validation (technically there's no timeout for adding or removing either, which is where you got this from I'm guessing -- should probably remove any references to a timeout in the real runtime class for now)